File: python/programmers/stack_queue/올바른 괄호.py
def solution(s):
    answer = True
    n = 0
    for w in s:
        if n == 0 and w == ')':
            return False
        elif w == '(':
            n += 1
        elif w == ')':
            n -= 1
    if n != 0:
        return False
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/올바른_괄호.java
package ps_traning.programmers.level_2;

import java.util.Stack;

public class 올바른_괄호 {
    boolean solution(String s) {
        boolean answer = true;
        Stack stk = new Stack();
        for (char c : s.toCharArray()) {
            if (c == '(') {
                stk.push(1);
            } else if (c == ')') {
                if (stk.size() == 0) {
                    answer = false;
                    break;
                }
                stk.pop();
            }
        }
        if (stk.size() != 0) {
            answer = false;
        }

        return answer;
    }
}
